GitCommands/Submodules/SubmoduleStatusProvider.cs

```diff
@@ -100,7 +100,7 @@ public void UpdateSubmodulesStatus(bool updateStatus, string workingDirectory, s
                 // First task: Gather list of submodules on a background thread.
 
                 // Don't access Module directly because it's not thread-safe.  Use a thread-local version:
-                var threadModule = new GitModule(workingDirectory);
+                var threadModule = new GitModule(gitModule);
```

Comment:
❗️ I don't like this very much, this may lead to more threading problems.

An instance of `GitModule` is not thread safe and it is a mixed bag of state and behaviours. That's why an instance of `GitModule` is either constructed for a given working directory or it is passed as `Func<IGitModule>`.

Reply:
This constructor copies static fields from the existing GitModule that do not need to be queried again, no threading issue

Reply:
Note: The second commit is not important, but it resolves one of the issues I investigated.

Reply:
I changed this, to make a simple copy, as used in other similar scenarios.

Reply:
Submitted #6373 for this